### PR TITLE
Change redirect responses to use HTTP 307 vs. HTTP 301

### PR DESCRIPTION
**Pull request by gideonthomas**

### CODE REVIEW

File: lib/middleware.js [hunk 1]
```diff
@@ -521,7 +521,7 @@ module.exports = function middlewareConstructor() {
       if(req.session.user) {
         next();
       } else {
-        res.redirect(301, "/anonymous/" + uuid.v1() + qs);
+        res.redirect(307, "/anonymous/" + uuid.v1() + qs);
```
Comment (humphd) [inline lib/middleware.js]: `v4()` vs. `v1()`

File: routes/projects/remix.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ module.exports = function(config, req, res) {
   var publishedId = req.params.publishedId;
   var user = req.user;
   if(!user) {
-    res.redirect("/anonymous/" + uuid.v1() + "/" + publishedId);
+    res.redirect(307, "/anonymous/" + uuid.v1() + "/" + publishedId);
```
Comment (humphd) [inline routes/projects/remix.js]: `v4()` vs. `v1()`

File: routes/projects/remix.js [hunk 3]
```diff
@@ -30,6 +30,6 @@ module.exports = function(config, req, res) {
       return;
     }
 
-    res.redirect(301, "/user/" + user.username + "/" + JSON.parse(body).id);
+    res.redirect(307, "/user/" + user.username + "/" + JSON.parse(body).id);
```
Comment (humphd) [inline routes/projects/remix.js]: Are you at all worried about `JSON.parse` throwing?

**Comment by humphd** [comment]
r=me with those things dealt with.

File: routes/projects/remix.js (hunk 3 above)
Comment (gideonthomas) [inline routes/projects/remix.js]: No coz publish is stable enough for us to trust the data it sends back

Comment (humphd) [inline routes/projects/remix.js]: I think that's a bad idea, but I won't block this patch on that.

**Pull request merged**